packages/@aws-cdk/toolkit-lib/lib/toolkit/non-interactive-io-host.ts

```diff
@@ -0,0 +1,174 @@
+import * as chalk from 'chalk';
+import type { IActivityPrinter } from '../api/shared-private';
+import { HistoryActivityPrinter, isMessageRelevantForLevel } from '../api/shared-private';
+import type { IIoHost, IoMessage, IoMessageLevel, IoRequest } from '../api/shared-public';
+import { isCI, isTTY } from '../util/shell-env';
+
+export interface NonInteractiveIoHostProps {
+  /**
+   * Determines the verbosity of the output.
+   *
+   * The IoHost will still receive all messages and requests,
+   * but only the messages included in this level will be printed.
+   *
+   * @default 'info'
+   */
+  readonly logLevel?: IoMessageLevel;
+
+  /**
+   * Overrides the automatic TTY detection.
+   *
+   * When TTY is disabled, the CLI will have no interactions or color.
+   *
+   * @default - determined from the current process
+   */
+  readonly isTTY?: boolean;
+
+  /**
+   * Whether the IoHost is running in CI mode.
+   *
+   * In CI mode, all non-error output goes to stdout instead of stderr.
+   * Set to false in the IoHost constructor it will be overwritten if the CLI CI argument is passed
+   *
+   * @default - determined from the environment, specifically based on `process.env.CI`
+   */
+  readonly isCI?: boolean;
+}
+
+/**
+ * A simple IO host for a non interactive CLI that writes messages to the console.
```

Comment:
...and returns the default answer to all queries?

Or fails all queries?

Reply:
>...and returns the default answer to all queries?

This. There's not really a concept of failing. For booleans you might consider the inverse a failure or it could just throw.